product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.1-registering-an-application/src/test/java/org/wso2/am/scenario/tests/register/application/ApplicationCreationTestCases.java

```diff
@@ -0,0 +1,210 @@
+    private static final String ERROR_GENERATING_SANDBOX_KEY = "Sandbox key generation failed for application:  ";
+    private static final String ERROR = "error";
+    private static final String STATUS = "status";
+    private static final String STATUS_APPROVED = "APPROVED";
+    private static final String NAME = "name";
+    private static final String TIER = "tier";
+    private static final String DESCRIPTION = "description";
+    private static final String TOKEN_TYPE = "tokenType";
+    private static final String APPLICATIONS = "applications";
+    private static final String DATA = "data";
+    private static final String KEY = "key";
+    private static final String KEY_STATE = "keyState";
+    private static final String APP_DETAILS = "appDetails";
+    private static final String KEY_TYPE = "key_type";
+    private static final String PRODUCTION = "PRODUCTION";
+    private static final String SANDBOX = "SANDBOX";
+    private static final String APPLICATION_KEY_GENERATION = "Application - key generation";
+    private static final String APPLICATION_DESCRIPTION = "Application description";
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws APIManagerIntegrationTestException {
+        String storeURL;
+        Properties infraProperties;
+
+        infraProperties = getDeploymentProperties();
+        storeURL = infraProperties.getProperty(STORE_URL);
+        if (storeURL == null) {
+            storeURL = DEFAULT_STORE_URL;
+        }
+        setKeyStoreProperties();
+        apiStore = new APIStoreRestClient(storeURL);
+        apiStore.login(ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+    }
+
+    @Test(description = "4.1.1.1", dataProvider = "ValidMandatoryApplicationValuesDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testApplicationCreationWithValidMandatoryValues(String applicationName, String tier, String description)
+            throws Exception {
+        HttpResponse addApplicationResponse = apiStore
+                .addApplication(URLEncoder.encode(applicationName, UTF_8), URLEncoder.encode(tier, UTF_8),
+                        "", URLEncoder.encode(description, UTF_8));
+        applicationsList.add(applicationName);
+        assertEquals(addApplicationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                ERROR_APPLICATION_CREATION_RESPONSE_CODE + applicationName);
+        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
+        assertFalse(addApplicationJsonObject.getBoolean(ERROR),
+                ERROR_APPLICATION_CREATION_WITH_VALID_INPUT + applicationName);
+        assertEquals(addApplicationJsonObject.get(STATUS), STATUS_APPROVED,
+                ERROR_APPLICATION_CREATION_WITH_VALID_INPUT + applicationName);
+        validateApplicationWithValidMandatoryValues(applicationName, tier, description);
+        destroy();
```

Comment:
Since i no longer need the applications i used that call to remove them. else shall i remove the call and let the AfterClass method call to handle it?

Reply:
Let the after class handle it. :)

Reply:
okay, fixed it